hwy/tests/logical_test.cc

```diff
@@ -226,6 +226,236 @@ HWY_NOINLINE void TestAllPopulationCount() {
   ForUnsignedTypes(ForPartialVectors<TestPopulationCount>());
 }
 
+template <class T, HWY_IF_NOT_FLOAT_NOR_SPECIAL(T), HWY_IF_T_SIZE(T, 4)>
+static HWY_INLINE T LeadingZeroCountOfValue(T val) {
+  const uint32_t u32_val = static_cast<uint32_t>(val);
+  return static_cast<T>(u32_val ? Num0BitsAboveMS1Bit_Nonzero32(u32_val) : 32);
+}
+template <class T, HWY_IF_NOT_FLOAT_NOR_SPECIAL(T), HWY_IF_T_SIZE(T, 8)>
+static HWY_INLINE T LeadingZeroCountOfValue(T val) {
+  const uint64_t u64_val = static_cast<uint64_t>(val);
+  return static_cast<T>(u64_val ? Num0BitsAboveMS1Bit_Nonzero64(u64_val) : 64);
+}
+template <class T, HWY_IF_NOT_FLOAT_NOR_SPECIAL(T),
+          HWY_IF_T_SIZE_ONE_OF(T, (1 << 1) | (1 << 2))>
+static HWY_INLINE T LeadingZeroCountOfValue(T val) {
+  using TU = MakeUnsigned<T>;
+  constexpr uint32_t kNumOfExtraLeadingZeros{32 - (sizeof(T) * 8)};
+  return static_cast<T>(
+      LeadingZeroCountOfValue(static_cast<uint32_t>(static_cast<TU>(val))) -
+      kNumOfExtraLeadingZeros);
+}
+
+struct TestLeadingZeroCount {
+  template <class T, class D>
+  HWY_NOINLINE void operator()(T /*unused*/, D d) {
+    RandomState rng;
+    using TU = MakeUnsigned<T>;
+    const RebindToUnsigned<decltype(d)> du;
+    size_t N = Lanes(d);
+    auto data = AllocateAligned<T>(N);
+    auto lzcnt = AllocateAligned<T>(N);
+
+    constexpr T kNumOfBitsInT{sizeof(T) * 8};
+    for (size_t j = 0; j < N; j++) {
+      lzcnt[j] = kNumOfBitsInT;
+    }
+    HWY_ASSERT_VEC_EQ(d, lzcnt.get(), LeadingZeroCount(Zero(d)));
```

Comment:
No need to change this but just FYI it is possible to `HWY_ASSERT_VEC_EQ(d, Set(d, kNumOfBitsInT), LeadingZeroCount())`.